source/fuzz/transformation_flatten_conditional_branch.cpp

```diff
@@ -52,6 +57,63 @@ bool TransformationFlattenConditionalBranch::IsApplicable(
     return false;
   }
 
+  std::set<uint32_t> used_fresh_ids;
+
+  // If ids have been provided to be used as vector guards for OpSelect
+  // instructions then they must be fresh.
+  for (uint32_t fresh_id_for_bvec_selector :
+       {message_.fresh_id_for_bvec2_selector(),
+        message_.fresh_id_for_bvec3_selector(),
+        message_.fresh_id_for_bvec4_selector()}) {
+    if (fresh_id_for_bvec_selector != 0) {
+      if (!CheckIdIsFreshAndNotUsedByThisTransformation(
+              fresh_id_for_bvec_selector, ir_context, &used_fresh_ids)) {
+        return false;
+      }
+    }
+  }
+
+  if (OpSelectArgumentsAreRestricted(ir_context)) {
```

Comment:
Could add a comment to describe what this next statement is doing.

Reply:
Done - and I commented that code a bit more in general.